tests/test_gdcm_pixel_data.py

```diff
@@ -0,0 +1,201 @@
+have_gdcm_handler = True
+try:
+    import pixel_data_handlers.gdcm_handler as gdcm_handler
+except ImportError:
+    have_gdcm_handler = False
+
+
+test_dir = os.path.dirname(__file__)
+test_files = os.path.join(test_dir, 'test_files')
+
+empty_number_tags_name = os.path.join(test_files, "reportsi_with_empty_number_tags.dcm")
+rtplan_name = os.path.join(test_files, "rtplan.dcm")
+rtdose_name = os.path.join(test_files, "rtdose.dcm")
+ct_name = os.path.join(test_files, "CT_small.dcm")
+mr_name = os.path.join(test_files, "MR_small.dcm")
+truncated_mr_name = os.path.join(test_files, "MR_truncated.dcm")
+jpeg2000_name = os.path.join(test_files, "JPEG2000.dcm")
+jpeg2000_lossless_name = os.path.join(test_files, "MR_small_jp2klossless.dcm")
+jpeg_ls_lossless_name = os.path.join(test_files, "MR_small_jpeg_ls_lossless.dcm")
+jpeg_lossy_name = os.path.join(test_files, "JPEG-lossy.dcm")
+jpeg_lossless_name = os.path.join(test_files, "JPEG-LL.dcm")
+deflate_name = os.path.join(test_files, "image_dfl.dcm")
+rtstruct_name = os.path.join(test_files, "rtstruct.dcm")
+priv_SQ_name = os.path.join(test_files, "priv_SQ.dcm")
+nested_priv_SQ_name = os.path.join(test_files, "nested_priv_SQ.dcm")
+meta_missing_tsyntax_name = os.path.join(test_files, "meta_missing_tsyntax.dcm")
+no_meta_group_length = os.path.join(test_files, "no_meta_group_length.dcm")
+gzip_name = os.path.join(test_files, "zipMR.gz")
+color_px_name = os.path.join(test_files, "color-px.dcm")
+color_pl_name = os.path.join(test_files, "color-pl.dcm")
+explicit_vr_le_no_meta = os.path.join(test_files, "ExplVR_LitEndNoMeta.dcm")
+explicit_vr_be_no_meta = os.path.join(test_files, "ExplVR_BigEndNoMeta.dcm")
+emri_name = os.path.join(test_files, "emri_small.dcm")
+emri_big_endian_name = os.path.join(test_files, "emri_small_big_endian.dcm")
+emri_jpeg_ls_lossless = os.path.join(test_files, "emri_small_jpeg_ls_lossless.dcm")
+emri_jpeg_2k_lossless = os.path.join(test_files, "emri_small_jpeg_2k_lossless.dcm")
+color_3d_jpeg_baseline = os.path.join(test_files, "color3d_jpeg_baseline.dcm")
+dir_name = os.path.dirname(sys.argv[0])
+save_dir = os.getcwd()
+
+
+class GDCM_JPEG_LS_Tests(unittest.TestCase):
+    def setUp(self):
+        self.jpeg_ls_lossless = read_file(jpeg_ls_lossless_name)
+        self.mr_small = read_file(mr_name)
+        self.emri_jpeg_ls_lossless = read_file(emri_jpeg_ls_lossless)
+        self.emri_small = read_file(emri_name)
+        self.original_handlers = pydicom.config.image_handlers
+        pydicom.config.image_handlers = [gdcm_handler]
+
+    def tearDown(self):
+        pydicom.config.image_handlers = self.original_handlers
+
+    @pytest.mark.skipif(have_gdcm_handler, reason="GDCM pixel data extension is being tested")
+    def test_JPEG_LS_PixelArray_no_gdcm(self):
+        with self.assertRaises((NotImplementedError, )):
+            _ = self.jpeg_ls_lossless.pixel_array
+
+    @pytest.mark.skipif(not have_gdcm_handler, reason="GDCM testing is being skipped")
```

Comment:
I think ```reason="GDCM is not available in this test environment"``` is a bit clearer.  Also since the string is repeated many times, maybe it should be a module-level constant, something like `no_gdcm_message` (similarly for the other message)